entity-management-solr/src/main/java/eu/europeana/entitymanagement/solr/model/SolrAgent.java

```diff
@@ -0,0 +1,99 @@
+package eu.europeana.entitymanagement.solr.model;
+
+import java.util.List;
+import java.util.Map;
+
+import eu.europeana.entitymanagement.definitions.model.Entity;
+import org.apache.solr.client.solrj.beans.Field;
+
+import eu.europeana.entitymanagement.definitions.model.Agent;
+import eu.europeana.entitymanagement.vocabulary.AgentSolrFields;
+import eu.europeana.entitymanagement.vocabulary.EntitySolrFields;
+
+/*
+ * TODO:see how to save the referencedWebResource and isAggregatedBy fields for all entities
+ */
+public class SolrAgent extends SolrEntity<Agent> {
+
+	@Field(AgentSolrFields.DATE)
+	private List<String> date;
+
+	@Field(AgentSolrFields.BEGIN)
+	private List<String> begin;
+
+	@Field(AgentSolrFields.END)
+	private List<String> end;
+
+	@Field(AgentSolrFields.DATE_OF_BIRTH_ALL)
+	private List<String> dateOfBirth;
+
+	@Field(AgentSolrFields.DATE_OF_DEATH_ALL)
+	private List<String> dateOfDeath;
+
+	@Field(AgentSolrFields.WAS_PRESENT_AT)
+	private List<String> wasPresentAt;
+
+	@Field(AgentSolrFields.HAS_MET)
+	private List<String> hasMet;
+
+	@Field(AgentSolrFields.NAME)
+	private Map<String, String> name;
+
+	@Field(AgentSolrFields.BIOGRAPHICAL_INFORMATION_ALL)
+	private Map<String, List<String>> biographicalInformation;
+
+	@Field(AgentSolrFields.PROFESSION_OR_OCCUPATION_ALL)
+	private List<String> professionOrOccupation;
+
+	@Field(AgentSolrFields.PLACE_OF_BIRTH_ALL)
+	private Map<String, List<String>> placeOfBirth;
+
+	@Field(AgentSolrFields.PLACE_OF_DEATH_ALL)
+	private Map<String, List<String>> placeOfDeath;
+
+	@Field(AgentSolrFields.DATE_OF_ESTABLISHMENT)
+	private String dateOfEstablishment;
+
+	@Field(AgentSolrFields.DATE_OF_TERMINATION)
+	private String dateOfTermination;
+
+	@Field(AgentSolrFields.GENDER)
+	private String gender;
+
+	public SolrAgent(Agent agent) {
+		super(agent);
+
+		
+		this.date = agent.getDate();
```

Comment:
aren't these values already set in the super(agent) ? Only the multilingual maps need preProcessing to append solrFieldName to labels, but that should also be applied automatically through the call of the supper implementation

Reply:
The new solr classes are implemented.